src/test/kotlin/org/rust/ide/typing/RsBraceMatcherTest.kt

```diff
@@ -8,49 +8,139 @@ package org.rust.ide.typing
 import com.intellij.codeInsight.highlighting.BraceMatchingUtil
 import com.intellij.codeInsight.highlighting.BraceMatchingUtil.getMatchedBraceOffset
 import com.intellij.openapi.editor.ex.EditorEx
+import org.intellij.lang.annotations.Language
 import org.rust.RsTestBase
-import org.rust.lang.RsFileType
 
 class RsBraceMatcherTest : RsTestBase() {
     fun `test don't pair parenthesis before identifier`() = doTest(
-        "fn main() { let _ = <caret>typing }",
+        "fn main() { let _ = /*caret*/typing; }",
         '(',
-        "fn main() { let _ = (<caret>typing }"
+        "fn main() { let _ = (/*caret*/typing; }"
+    )
+
+    fun `test don't pair parenthesis before string literal`() = doTest(
+        """fn main() { let _ = /*caret*/"foo"; }""",
+        '(',
+        """fn main() { let _ = (/*caret*/"foo"; }"""
+    )
+
+    fun `test don't pair parenthesis before integral literal`() = doTest(
+        "fn main() { let _ = /*caret*/1; }",
+        '(',
+        "fn main() { let _ = (/*caret*/1; }"
+    )
+
+    fun `test don't pair parenthesis before ref operator`() = doTest(
+        "fn main() { let _ = /*caret*/&foo; }",
+        '(',
+        "fn main() { let _ = (/*caret*/&foo; }"
+    )
+
+    fun `test don't pair parenthesis before dereference operator`() = doTest(
+        "fn main() { let _ = /*caret*/*foo.bar; }",
+        '(',
+        "fn main() { let _ = (/*caret*/*foo.bar; }"
+    )
+
+    fun `test don't pair parenthesis before not operator`() = doTest(
+        "fn main() { let _ = /*caret*/!foo; }",
+        '(',
+        "fn main() { let _ = (/*caret*/!foo; }"
+    )
+
+    fun `test don't pair parenthesis before unary minus operator`() = doTest(
+        "fn main() { let _ = /*caret*/-foo; }",
+        '(',
+        "fn main() { let _ = (/*caret*/-foo; }"
+    )
+
+    fun `test don't pair parenthesis before box operator`() = doTest(
+        "fn main() { let _ = /*caret*/box foo; }",
+        '(',
+        "fn main() { let _ = (/*caret*/box foo; }"
+    )
+
+    fun `test don't pair parenthesis before if expression`() = doTest(
+        "fn main() { let _ = /*caret*/if foo { bar } else { baz }; }",
+        '(',
+        "fn main() { let _ = (/*caret*/if foo { bar } else { baz }; }"
+    )
+
+    fun `test don't pair parenthesis before match expression`() = doTest(
+        "fn main() { let _ = /*caret*/match foo { _ => () }; }",
+        '(',
+        "fn main() { let _ = (/*caret*/match foo { _ => () }; }"
+    )
```

Comment:
Probably should also test for `{` and `[`. `[]/{}` are syntactically invalid before `if` and `match`, and the user may want to enclose those expressions in a block expression or an array.